metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1583,7 +1585,9 @@ class MetalsLanguageServer(
             definitionResult(position, token)
           }
         case None =>
-          warnings.noSemanticdb(source)
+          if (semanticDBDoc.isEmpty) {
```

Comment:
or we could do:
```scala
val semanticDBDoc = semanticdbs.textDocument(source).documentIncludingStale
if(semanticDBDoc.isEmpty){
  warnings.noSemanticdb(source)
} else {
  val occurrence = for { ...} yield occ
  occurrence match {
    case None =>
      // ignore
    case Some(occ) =>
      (...)
  }
}
```
would probably make the logic simpler (no need to remember about semanticDBDoc being emoty or not when working on occurrences).

or even something like that:

```scala

val semanticDBDoc = semanticdbs.textDocument(source).documentIncludingStale 
semanticDBDoc match {
  case None => warnings.noSemanticdb(source)
  case Some(doc) =>
    val occurrence = definitionProvider.positionOccurrence(source, position, doc).occurrence 
    occurrence.foreach { occ =>
        (...)
    }
```

Reply:
I'd rather not change too much now.